src/bloqade/squin/rewrite/wrap_analysis.py

```diff
@@ -40,22 +41,11 @@ def print_impl(self, printer: Printer) -> None:
 
 
 @dataclass
-class WrapSquinAnalysis(RewriteRule):
-
-    address_analysis: dict[ir.SSAValue, Address]
-    op_site_analysis: dict[ir.SSAValue, Sites]
+class WrapAnalysis(RewriteRule):
 
+    @abstractmethod
     def wrap(self, value: ir.SSAValue) -> bool:
-        address_analysis_result = self.address_analysis[value]
-        op_site_analysis_result = self.op_site_analysis[value]
-
-        if value.hints.get("address") and value.hints.get("sites"):
-            return False
-        else:
-            value.hints["address"] = AddressAttribute(address_analysis_result)
-            value.hints["sites"] = SitesAttribute(op_site_analysis_result)
-
-        return True
+        pass
 
     def rewrite_Block(self, node: ir.Block) -> RewriteResult:
         has_done_something = False
```

Comment:
do you need this? You should just use a `Walk` and only implement the `rewrite_Statement` method since the analysis results are referenced by the SSAValues.